staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/BatchPathProvider.java

```diff
@@ -109,6 +126,68 @@ public Path getTenantInprogressDirectory(final TenantId tenantId)
+        try (Stream<Path> fileStream = Files.list(pathForBatches)) {
+            fileStream.filter(batch -> BatchNameProvider.getBatchId(batch.getFileName().toString()).equals(batchId.getValue()) &&
+                            !batch.getFileName().toString().contains(ServiceIdentifier.getServiceId()))
+                    .map(batch -> batch.resolve("files"))
+                    .forEach(batch -> {
+                        try (Stream<Path> filePath = Files.walk(batch)) {
+                            size[0] = filePath.mapToLong(file -> file.toFile().length()).sum();
+                        } catch (IOException e) {
+                            LOGGER.info("Folder might have been deleted or moved.");
+                        }
+                        try (Stream<Path> files = Files.list(batch)) {
+                            Optional<Path> lastModifiedFile = files.max(Comparator.comparing(BatchPathProvider::getLastModified));
+                            if (lastModifiedFile.isPresent()) {
+                                Path file = lastModifiedFile.get();
+                                Map<String, Tracker> trackerMap = BatchProgressTracker.getInProgressTrackerMap();
+                                /*Check the file system and see differences between last modified time
+                                by making thread sleep for 1 second to confirm if upload is progressing*/
+                                Instant prevTime = getLastModified(file);
+                                threadSleep();
+                                Instant currTime = getLastModified(file);
+                                if (ChronoUnit.MILLIS.between(prevTime, currTime) > 0) {
+                                    Tracker tracker = new Tracker();
+                                    tracker.setLastModifiedTime(currTime);
+                                    tracker.setNumberOfBytesReceived(size[0]);
+                                    trackerMap.put(threadID, tracker);
+                                }
+                            }
+                        } catch (IllegalStateException | IOException e) {
+                            LOGGER.info("The file might have been moved to Completed.");
+                        }
+                    });
+        } catch (IOException e) {
+            LOGGER.info("The file might have been moved to Completed.");
+        }
+    }
+
+    private static Instant getLastModified(Path p)
+    {
+        try {
+            return Files.readAttributes(p, BasicFileAttributes.class).lastModifiedTime().toInstant();
+        } catch (IOException ioe) {
+            throw new IllegalStateException(ioe);
+        }
+    }
+
+    public List<String> getListOfInProgressThreadFromFileSystem(TenantId tenantId, BatchId batchId)
+    {
+        Path inProgressPath = getTenantInprogressDirectory(tenantId);
+        List<String> list = new ArrayList<>();
+        try (Stream<Path> stream = Files.list(inProgressPath)) {
```

Comment:
Variables declared in try-with-resources should also be `final`:

```suggestion
        try (final Stream<Path> stream = Files.list(inProgressPath)) {
```

Reply:
Done.